src/mito2/src/read/unordered_scan.rs

```diff
@@ -94,34 +95,66 @@ impl UnorderedScan {
         part_metrics: PartitionMetrics,
         range_builder_list: Arc<RangeBuilderList>,
     ) -> impl Stream<Item = Result<Batch>> {
-        stream! {
+        async fn build_scan_stream(
+            stream_ctx: Arc<StreamContext>,
+            index: RowGroupIndex,
+            time_range: FileTimeRange,
+            part_metrics: PartitionMetrics,
+            range_builder_list: Arc<RangeBuilderList>,
+        ) -> Result<BoxedBatchStream> {
+            let stream = if stream_ctx.is_mem_range_index(index) {
+                scan_mem_ranges(stream_ctx, part_metrics, index, time_range).boxed()
+            } else if stream_ctx.is_file_range_index(index) {
+                scan_file_ranges(
+                    stream_ctx,
+                    part_metrics,
+                    index,
+                    "unordered_scan_files",
+                    range_builder_list,
+                )
+                .await?
+                .boxed()
+            } else {
+                #[cfg(feature = "enterprise")]
+                {
+                    crate::read::scan_util::scan_extension_range(
+                        stream_ctx.clone(),
+                        index,
+                        part_metrics.clone(),
+                    )
+                    .await?
+                }
+                #[cfg(not(feature = "enterprise"))]
+                return crate::error::UnexpectedSnafu {
+                    reason: "unknown scan range type",
+                }
+                .fail();
+            };
+            Ok(stream)
+        }
```

Comment:
Looks like we don't have to wrap a function if we can simplify the usage of scan_extension_range.